src/bert/utils.py

```diff
@@ -24,6 +26,16 @@ def __init__(self, args, cache_dir=None):
         for param in self.model.parameters():
             param.requires_grad = bool(args.bert_fine_tune)
 
+        # Configure scalar mixing, ELMo-style
+        # TODO: if doing multiple target tasks, allow for multiple sets of
+        # scalars. Seee the ELMo implementation here:
+        # https://github.com/allenai/allennlp/blob/master/allennlp/modules/elmo.py#L115
+        if self.embeddings_mode == "mix":
+            num_layers = self.model.config.num_hidden_layers
+            self.scalar_mix = scalar_mix.ScalarMix(num_layers + 1,
+                                                   do_layer_norm=False)
```

Comment:
Are gradients to these handled in the same way as ELMo scalar gradients? (i.e., trained even when the rest of BERT isn't?)

Reply:
Yes, that's the intent. Added comment to clarify.

Reply:
Cool. I'll assume you have a way to double check this.